modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectLayoutLocalServiceTest.java

```diff
@@ -533,33 +545,64 @@ private ObjectLayoutBox _addObjectLayoutBox(String type) throws Exception {
 		objectLayoutBox.setType(type);
 
 		if (StringUtil.equals(type, ObjectLayoutBoxConstants.TYPE_REGULAR)) {
-			objectLayoutBox.setObjectLayoutRows(
+			List<ObjectLayoutRow> objectLayoutRows = new ArrayList<>(
 				Arrays.asList(
 					_addObjectLayoutRow(), _addObjectLayoutRow(),
 					_addObjectLayoutRow()));
+
+			if (!objectFieldIds.isEmpty()) {
+				objectLayoutRows.add(
+					_addObjectLayoutRow(
+						ListUtil.toList(
+							objectFieldIds, this::_addObjectLayoutColumn)));
+			}
+
+			objectLayoutBox.setObjectLayoutRows(objectLayoutRows);
 		}
 
 		return objectLayoutBox;
 	}
 
+	private ObjectLayoutBox _addObjectLayoutBoxWithSystemObjectFields()
+		throws Exception {
+
+		ObjectField objectField = _objectFieldLocalService.getObjectField(
+			_objectDefinition.getObjectDefinitionId(), "id");
+
+		return _addObjectLayoutBox(
+			Arrays.asList(
+				objectField.getObjectFieldId(), _addSystemObjectField()),
+			ObjectLayoutBoxConstants.TYPE_REGULAR);
+	}
```

Comment:
ID is a metadata field. Is it allowed to add metadata fields in the Layout?
I would prefer to add 2 system object fields in the test, like:
`Arrays.asList(_addSystemObjectField(), _addSystemObjectField())`

Reply:
Yeah, it's possible to add metadata field in the Layout.
In the story we allowed this, they appear as read-only in Layouts

Reply:
[Story acceptance criteria](https://liferay.atlassian.net/browse/LPS-193360):

```
2. Given an admin user
      When creating layouts and views for modifiable system objects
      Then I must be able to use system fields as well
      Metadata fields must be read-only in layouts (besides ERC)
```

Reply:
My bad, I didn't know about this.